packages/cli/src/runners/task-broker.service.ts

```diff
@@ -223,6 +233,14 @@ export class TaskBroker {
 		}
 	}
 
+	handleRunnerDeferred(taskId: Task['id']) {
+		const acceptReject = this.runnerAcceptRejects.get(taskId);
+		if (acceptReject) {
+			acceptReject.reject(new TaskDeferredError());
+			this.runnerAcceptRejects.delete(taskId);
+		}
```

Comment:
I don't think we actually need to reject here. We could push the task directly back to pending tasks

Reply:
Gave this a try and it kind of works, but I'm not satisfied - we need to add more properties to messages on both sides, add an exception to a timeout, and also something about this is [preventing main from exiting cleanly](https://share.cleanshot.com/26MRtMFV).

See these branches:

- `main/handshake-tomi-feedback`
- `launcher/handshake-tomi-feedback`

Let's discuss tomorrow.

Reply:
> something about this is [preventing main from exiting cleanly](https://share.cleanshot.com/26MRtMFV)

This is actually broken on `master` → https://github.com/n8n-io/n8n/pull/11801

Reply:
Oh right you are absolutely right. I didn't realize that `acceptOffer` does a bunch of stuff that must be handled as well. The way you implemented it is actually really nice of handling this 👌